tests/processor.test.js

```diff
@@ -38,792 +40,898 @@ describe("processor", () => {
-			]);
-
-			assert.strictEqual(result.length, 3);
-			assert.deepStrictEqual(result[0], {
-				...message,
-				line: 3,
-				column: 1,
-			});
-			assert.deepStrictEqual(result[1], {
-				...message,
-				line: 14,
-				column: 4,
-			});
-			assert.deepStrictEqual(result[2], {
-				...message,
-				line: 23,
-				column: 3,
+					[
+						{
+							line: 3,
+							endLine: 6,
+							column: 2,
+							message: "Unreachable code after return.",
+							ruleId: "no-unreachable",
+						},
+						{
+							line: 4,
+							endLine: 4,
+							column: 2,
+							message: "Unnecessary semicolon.",
+							ruleId: "no-extra-semi",
+							fix: { range: [38, 39], text: "" },
+						},
+					],
+				];
+
+				beforeEach(() => {
+					processor.preprocess(code);
+				});
+
+				it("should allow for no messages", () => {
+					const result = processor.postprocess([[], [], []]);
+
+					assert.strictEqual(result.length, 0);
+				});
+
+				it("should flatten messages", () => {
+					const result = processor.postprocess(messages);
+
+					assert.strictEqual(result.length, 5);
+					assert.strictEqual(
+						result[0].message,
+						'Use the global form of "use strict".',
+					);
+					assert.strictEqual(
+						result[1].message,
+						"Unexpected console statement.",
+					);
+					assert.strictEqual(
+						result[2].message,
+						"Missing trailing comma.",
+					);
+					assert.strictEqual(
+						result[3].message,
+						"Unreachable code after return.",
+					);
+					assert.strictEqual(
+						result[4].message,
+						"Unnecessary semicolon.",
+					);
+				});
+
+				it("should translate line numbers", () => {
+					const result = processor.postprocess(messages);
+
+					assert.strictEqual(result[0].line, 4);
+					assert.strictEqual(result[1].line, 6);
+					assert.strictEqual(result[2].line, 17);
+					assert.strictEqual(result[3].line, 26);
+					assert.strictEqual(result[4].line, 27);
+				});
+
+				it("should translate endLine numbers", () => {
+					const result = processor.postprocess(messages);
+
+					assert.strictEqual(result[0].endLine, 4);
+					assert.strictEqual(result[1].endLine, 6);
+					assert.strictEqual(result[2].endLine, 17);
+					assert.strictEqual(result[3].endLine, 29);
+					assert.strictEqual(result[4].endLine, 27);
+				});
+
+				it("should translate column numbers", () => {
+					const result = processor.postprocess(messages);
+
+					assert.strictEqual(result[0].column, 1);
+					assert.strictEqual(result[1].column, 5);
+				});
+
+				it("should translate indented column numbers", () => {
+					const result = processor.postprocess(messages);
+
+					assert.strictEqual(result[2].column, 9);
+					assert.strictEqual(result[3].column, 4);
+					assert.strictEqual(result[4].column, 2);
+				});
+
+				it("should adjust fix range properties", () => {
+					const result = processor.postprocess(messages);
+
+					assert(result[2].fix.range, [185, 185]);
+					assert(result[4].fix.range, [264, 265]);
+				});
```

Comment:
I'm not sure how, but this test was passing with BOM before the code change.

Reply:
Oh, this test was missing `.strictEqual` after `assert`.

Reply:
Ok, in https://github.com/eslint/markdown/pull/282/commits/68aede7ffaae6e3d8bcad59b8adec3cce85a51fa I fixed this test and added another one as per the repro from the original post. Both would fail without the code change.